File: main/com/example/service/calc/CalculationOperations.java
package com.example.service.calc;

import com.example.model.OperationModel;

public interface CalculationOperations {

    String add(OperationModel model);

    String subtract(OperationModel model);

    String multiply(OperationModel model);

    String divide(OperationModel model);

    String factorial(OperationModel model);

    String fibonacci(OperationModel model);

    String sqrt(OperationModel model); 

    String power(OperationModel model); 

    OperationModel resetCalculator(OperationModel model); 

    OperationModel resetAdvancedCalculator(OperationModel model);
}

File: main/com/example/service/calc/CalculationOperations.kt
package com.example.service.calc

import com.example.model.OperationModel

interface CalculationOperations {
    fun add(model: OperationModel): String
    fun subtract(model: OperationModel): String
    fun multiply(model: OperationModel): String
    fun divide(model: OperationModel): String
    fun factorial(model: OperationModel): String
    fun fibonacci(model: OperationModel): String
    fun sqrt(model: OperationModel): String
    fun power(model: OperationModel): String
    fun resetCalculator(model: OperationModel): OperationModel
    fun resetAdvancedCalculator(model: OperationModel): OperationModel
}